### ISSUE TITLE
[a11y] 9.3.3.1 - Error detection

### ISSUE THREAD

**Issue by tbsbdr** [owncloud/web#5396]
- Context: The Web UI should be accessible according to the WCAG2.1 /BITV 2.0 guidelines. The number in the title relates to the test-steps: https://www.bitvtest.de/bitv_test/das_testverfahren_im_detail/pruefschritte.html
- "Status" relates to the evaluation scheme https://www.bitvtest.de/bitv_test/das_testverfahren_im_detail/verfahren.html#c1262 
- "a11y" is short for accessibility
- ----------------
## Status: partly fulfilled
- **Page 1 (Alle Dateien)**
When creating a public link and in case the user leaves the fields empty, an error message appears above the form:
**a)** The error message is in English language but should be in German - "error creating public share"
**b)** The text of the error message is not informative because it doesn’t explain what went wrong (what data caused the error?)
**c)** The error message is not handled by screenreaders. One solution could be the use of aria- live or role="alert“.

## Status: fulfilled
- **Page 2 (Mit anderen geteilten Dateien)**

## Status: not applicable
- **Page 3 (Gelöschte Dateien)**
- **Page 4 (Konto)**

**Comment by paulcod3** [comment]
1. The workflow as described doesn't lead to an error.
to a) in the filesharing sidebar all messages are translated using $gettext
to b) as discribed in 1. the error doesn't exist, other possible errors in the component are meaningful
to c) role=alert will be set

**Comment by kulmann** [comment]
Error message is coming untranslated from the server.

**Comment by kulmann** [comment]
`OcAlert` has `role="alert"` now since https://github.com/owncloud/owncloud-design-system/pull/1530/commits/895c0908c44b6fa33ae6fddfc0ec4bde99b9895c

**Comment by kulmann** [comment]
We fixed the frontend side form validation for required password. Other fields don't have any requirements. Only thing left are unforeseen server side errors. For those we now show a generic error message as a notification (read by screenreaders, disappears after a few seconds). https://github.com/owncloud/web/pull/5512/commits/14a12b8ec7898289297f818c83a23b9a72d6499b

**Comment by kulmann** [comment]
Also made sure that the same is fine for people shares.
https://github.com/owncloud/web/pull/5512/commits/49c9f03e7e2c6ad5139aaa228b59bb3c3bed31c5
